.github/workflows/dsp_ar_build.yaml

```diff
@@ -0,0 +1,100 @@
+name: DSP GAR - Tag, Build and Push Image
+
+on:
+  push:
+    branches:
+    - develop
+  pull_request:
+    branches:
+    - develop
+env:
+  REGISTRY_HOST: us-central1-docker.pkg.dev
+  GOOGLE_PROJECT: dsp-artifact-registry
+  SERVICE_NAME: ${{ github.event.repository.name }}
+jobs:
+  tag-build-push:
+    runs-on: ubuntu-latest
+    outputs:
+      sherlock-version: ${{ steps.short-sha.outputs.sha }}
+    permissions:
+      contents: 'read'
+      id-token: 'write'
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v4
+      with:
+        persist-credentials: false
+    - name: Get Short Sha
+      id: short-sha
+      run: echo "sha=$(git rev-parse --short=12 HEAD)" >> $GITHUB_OUTPUT
+    - name: 'Set up Cloud SDK'
+      uses: 'google-github-actions/setup-gcloud@v2'
+    - name: Construct tags
+      id: construct-tags
+      run: |
+        SHA_TAG="${REGISTRY_HOST}/${GOOGLE_PROJECT}/${SERVICE_NAME}/${SERVICE_NAME}:${SHORT_SHA}"
+        ENVIRONMENT_TAG=""
+        if ${{ github.event_name == 'pull_request'}}; then
```

Comment:
I think to avoid a possible GitHub template injection, we need to pull out `${{ github.event_name }}` into an environment variable and then do a regular bash check like `if [ "$GITHUB_EVENT" = "pull_request" ]; then`